docs/build/advanced-concepts/randomness.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Flow's Native Onchain Randomness
+sidebar_label: Native Randomness
+---
+
+# Native Onchain Randomness on Flow
+
+Flow enhances blockchain functionality by providing native onchain randomness, eliminating reliance on external oracles in Web3. This secure, decentralized feature empowers developers to build a variety of applications with truly unpredictable, transparent, and fair outcomes, achieved with greater efficiency.
+
+Flow’s onchain randomness delivers immediate random values within smart contracts, bypassing the latency and complexity of oracle integration. Developers can obtain verifiably random results with a single line of Cadence code, streamlining the development process and enhancing the performance of decentralized applications.
+
+## Use Cases of Onchain Randomness
+
+- **Gaming:** Integrates fairness and unpredictability into gameplay, enhancing user engagement without delays.
+- **NFTs:** Facilitates the creation of uniquely randomized traits in NFTs quickly, adding to their rarity and value.
+- **Lotteries & Draws:** Offers instant and verifiably fair random selection for lotteries, solidifying trust in real-time.
+- **DeFi Protocols:** Enables rapid and innovative random reward systems within decentralized finance.
+- **DAOs:** Assists in unbiased voting and task assignments through immediate randomness.
+- **Broad Applications:** Extends to any domain requiring impartial randomization, from asset distribution to security mechanisms, all with the added benefit of on-demand availability.
+
+## History - Distributed Random Beacon
+
+Within the Flow protocol, the heart of randomness generation lies in the "distributed random beacon." This module generates randomness that is distributed across the network while adhering to established safety and security standards. The output from the randomness beacon is a sequence of random bytes that are unpredictable and impartial.
+
+For over three years, the beacon has ensured protocol security by assigning verification nodes to oversee block computations. For those interested in a more detailed exploration of the randomness beacon and its inner workings, you can read [the technical deep dive on the Flow forum](https://forum.flow.com/t/secure-random-number-generator-for-flow-s-smart-contracts/5110).
+
+## Guidelines for Safe Usage
+
+For usage of randomness where result abortion is not an issue, it is recommended to use the Cadence built-in function `revertibleRandom.` `revertibleRandom` returns a pseudo-random number and is also based on the safe Random Beacon.
+
+```cadence
+// Language reference:
```

Comment:
Could also be nice to link to the ecdao snippet: https://academy.ecdao.org/en/snippets/cadence-random